m2cgen/assemblers/linear.py

```diff
@@ -10,19 +9,17 @@ def assemble(self):
         return self._build_ast()
 
     def _build_ast(self):
-        coef = self.model.coef_
-        intercept = self.model.intercept_
-        if isinstance(coef, np.ndarray) and len(coef.shape) == 2:
-            if coef.shape[0] == 1:
-                return _linear_to_ast(coef[0], intercept[0])
-            else:
-                exprs = []
-                for idx in range(coef.shape[0]):
-                    exprs.append(ast.SubroutineExpr(
-                        _linear_to_ast(coef[idx], intercept[idx])))
-                return ast.ArrayExpr(exprs)
+        coef = utils.to_2d_array(self.model.coef_)
+        intercept = utils.to_2d_array(self.model.intercept_)
```

Comment:
Why do we need to transform this to 2d array? We always expect either scalar or a 1d array here.

Reply:
Fixed